Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -21,16 +21,23 @@ function SetTaskContextVariable
     param([string][ValidateNotNullOrEmpty()]$varName, 
           [string]$varValue)
         
-    #[Environment]::SetEnvironmentVariable($varName, $varValue)
+    [Environment]::SetEnvironmentVariable($varName, $varValue)
     Write-Host "##vso[task.setvariable variable=$varName;]$varValue"
 }
 
 function GetTaskContextVariable()
 {
 	param([string][ValidateNotNullOrEmpty()]$varName)
         
-    #return ([Environment]::GetEnvironmentVariable($varName))
-	return Get-TaskVariable -Context $distributedTaskContext -Name $varName    
+    $value = [Environment]::GetEnvironmentVariable($varName);
```

Comment:
Hi Pascal,

The docs for Get-TaskVariable and Set-TaskVariable state that the variables will be available for use in other steps. However, the classical agent (not the .net core one) uses a single process to execute all build steps, so the variables were available everywhere. The .net core agent uses separate processes.

Bogdan

From: Pascal Berger [mailto:notifications@github.com]
Sent: Wednesday, July 20, 2016 10:25 PM
To: Microsoft/vsts-tasks <vsts-tasks@noreply.github.com>
Cc: Bogdan Gavril <bgavril@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/vsts-tasks] Users/bgavril/revert (#2108)


In Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1<https://github.com/Microsoft/vsts-tasks/pull/2108#discussion_r71609689>:

>      Write-Host "##vso[task.setvariable variable=$varName;]$varValue"

>  }

>

>  function GetTaskContextVariable()

>  {

>       param([string][ValidateNotNullOrEmpty()]$varName)

>

> -    #return ([Environment]::GetEnvironmentVariable($varName))

> -     return Get-TaskVariable -Context $distributedTaskContext -Name $varName

> +    $value = [Environment]::GetEnvironmentVariable($varName);

@bgavrilMS<https://github.com/bgavrilMS> Isn't this a bug in the agent and should be fixed there?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/vsts-tasks/pull/2108/files/bad40d791ea32f421d6ef39f8603855d9f08c2a5..ae267651f15156b0654038831b0562ec85c84dd1#r71609689>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ALtG6BMH5Thf1N9YpdcKicHhoy6DJ9X7ks5qXpIvgaJpZM4JQqsg>.